src/services/formatting/formatting.ts

```diff
@@ -739,8 +739,7 @@ namespace ts.formatting {
                     // if child node is a token, it does not impact indentation, proceed it using parent indentation scope rules
                     const tokenInfo = formattingScanner.readTokenInfo(child);
                     // JSX text shouldn't affect indenting
-                    if (child.kind !== SyntaxKind.JsxText) {
-                        Debug.assert(tokenInfo.token.end === child.end, "Token end is child end");
+                    if (child.kind !== SyntaxKind.JsxText && tokenInfo.token.end === child.end) {

```

Comment:
I don't fully understand why `token.end` and `child.end` needs to match, but this change has no side effect other than fixing the bug. 

If any of you know how this works I can dig deeper for any unexpected results.